ir_datasets/datasets/trec_fair_2021.py

```diff
@@ -0,0 +1,179 @@
+                        yield FairTrecDoc(data1['id'], data1['title'],data1['text'],plaintext, data1['url'], str(match['quality_score']), match['geographic_locations'], str(match['quality_score_disc']))
+                    else: 
+                   
+                        plaintext = ""
+                        plaintext = textifier.textify(data1['text'])
+                        yield FairTrecDoc(data1['id'], data1['title'],data1['text'],plaintext, data1['url'], "NA","NA","NA")
+                except ValueError as err:
+                    print("[{} - {}]: {}".format(data1['id'], data1['title'], err))
+
+
+  
+              
+
+    def docs_cls(self):
+        return FairTrecDoc
+
+    def docs_store(self, field='doc_id'):
+        return PickleLz4FullStore(
+            path=f'{ir_datasets.util.home_path()/NAME/self._name}/docs.pklz4',
+            init_iter_fn=self._docs_iter,
+            data_cls=self.docs_cls(),
+            lookup_field=field, 
+            index_fields=['doc_id']
+        )
+
+    def docs_count(self):
+        return self.docs_store().count()
+
+    def docs_namespace(self):
+        return f'{NAME}/{self._name}'
+
+    def docs_lang(self):
+        return 'en'
+
+
+class FairTrecQueries(BaseQueries):
+    def __init__(self, name, dlc, keep_metadata=None):
+        super().__init__()
+        self._name = name
+        self._dlc = dlc
+        self._keep_metadata = keep_metadata
+
+    def queries_iter(self):
+        with self._dlc.stream() as stream:
+            for line in stream:
+                data = json.loads(line)
+                yield FairTrecQuery(data['id'], data['title'])
+
+    def queries_cls(self):
+        return FairTrecQuery
+
+    def queries_namespace(self):
+        return f'{NAME}/{self._name}'
+
+    def queries_lang(self):
+        return 'en'
+
+
+class FairTrecQrels(BaseQrels):
+    def __init__(self, qrels_dlc):
+        self._qrels_dlc = qrels_dlc
+       
+
+    def qrels_path(self):
+        return self._qrels_dlc.path()
+
+
+    def qrels_iter(self):
+        with self._qrels_dlc.stream() as stream:
+            for line in stream:
+                data = json.loads(line)
+                for rlDoc in data["rel_docs"]:
+                    yield TrecQrel(data["id"], rlDoc, 1,0)
+
+
+
+    def qrels_cls(self):
+        return TrecQrel
+
+    def qrels_defs(self):
+        return self._qrels_defs
+
+
+def _init():
+    base_path = ir_datasets.util.home_path()/NAME
+    dlc = ir_datasets.util.DownloadConfig.context(NAME, base_path)
+    documentation = YamlDocumentation(f'docs/{NAME}.yaml')
+
+    base = Dataset(documentation('_'))
+
+  
+    dataset = Dataset(
+            FairTrecDocs('docs', GzipExtract(dlc["docs"]), GzipExtract(dlc["metadata"])),
+            FairTrecQueries('queries', Cache(GzipExtract(dlc["topics"]), base_path/'trec_fair_2021'/'queries.json')),
+            FairTrecQrels(GzipExtract(dlc["topics"])),
+            documentation('trec_fair_2021'))
```

Comment:
I think this needs to be '_' to link up properly with the key in `ir_datasets/docs/trec_fair_2021.yaml`